synapse/handlers/relations.py

```diff
@@ -270,16 +267,34 @@ async def get_threads_for_events(
         """
         user = UserID.from_string(user_id)
 
+        # It is not valid to start a thread on an event which itself relates to another event.
+        event_ids = [eid for eid in events_by_id.keys() if eid not in relations_by_id]
+
         # Fetch thread summaries.
         summaries = await self._main_store.get_thread_summaries(event_ids)
 
-        # Only fetch participated for a limited selection based on what had
+        # Only fetch whether the current user has participated in a thread based
+        # on which returned for a limited selection based on what had
         # summaries.
         thread_event_ids = [
             event_id for event_id, summary in summaries.items() if summary
         ]
-        participated = await self._main_store.get_threads_participated(
-            thread_event_ids, user_id
+
+        # Preseed thread participation with whether the requester send the event.
```

Comment:
```suggestion
        # Pre-seed thread participation with whether the requester sent the event.
```

Reply:
Maybe if these are always root thread events:

```suggestion
        # Pre-seed thread participation with whether the requester sent the root event.
```